### PR TITLE
Debounce changes

### PR DESCRIPTION
**Pull request by rohilshahZC**

### CODE REVIEW

File: zoomdls/src/main/java/com/zoomcar/uikit/location/ZLocationCell.kt [hunk 1]
```diff
@@ -33,9 +36,19 @@ class ZLocationCell : ConstraintLayout {
 
         override fun afterTextChanged(s: Editable?) {
             if (binding.textPickupLocation.hasFocus()) {
-                listener?.onLocationSearchTextChanged(s.toString(), LocationSearchFlowType.PICKUP)
+                listener?.onLocationSearchTextChanged(s.toString(), LocationSearchFlowType.PICKUP, false)
+                debounceHandler.removeCallbacksAndMessages(null)
+                debounceHandler.postDelayed({
+                    listener?.onLocationSearchTextChanged(s.toString(), LocationSearchFlowType.PICKUP, true)
+                },
+                        1000)
```
Comment (shishir5) [inline zoomdls/src/main/java/com/zoomcar/uikit/location/ZLocationCell.kt:44]: define the constant above

File: zoomdls/src/main/java/com/zoomcar/uikit/location/ZLocationCell.kt [hunk 2]
```diff
@@ -33,9 +36,19 @@ class ZLocationCell : ConstraintLayout {
...
             } else if (binding.textDropOffLocation.hasFocus()) {
-                listener?.onLocationSearchTextChanged(s.toString(), LocationSearchFlowType.DROP_OFF)
+                listener?.onLocationSearchTextChanged(s.toString(), LocationSearchFlowType.DROP_OFF, false)
+                debounceHandler.removeCallbacksAndMessages(null)
+                debounceHandler.postDelayed({
+                    listener?.onLocationSearchTextChanged(s.toString(), LocationSearchFlowType.DROP_OFF, true)
```
Comment (shishir5) [inline zoomdls/src/main/java/com/zoomcar/uikit/location/ZLocationCell.kt:49]: can use apply

File: zoomdls/src/main/java/com/zoomcar/uikit/location/ZLocationCell.kt [hunk 3]
```diff
@@ -33,9 +36,19 @@ class ZLocationCell : ConstraintLayout {
 
         override fun afterTextChanged(s: Editable?) {
             if (binding.textPickupLocation.hasFocus()) {
-                listener?.onLocationSearchTextChanged(s.toString(), LocationSearchFlowType.PICKUP)
+                listener?.onLocationSearchTextChanged(s.toString(), LocationSearchFlowType.PICKUP, false)
+                debounceHandler.removeCallbacksAndMessages(null)
+                debounceHandler.postDelayed({
+                    listener?.onLocationSearchTextChanged(s.toString(), LocationSearchFlowType.PICKUP, true)
```
Comment (shishir5) [inline zoomdls/src/main/java/com/zoomcar/uikit/location/ZLocationCell.kt:42]: can use apply

**Review by VaiSandeep — APPROVED** [review]

File: zoomdls/src/main/java/com/zoomcar/uikit/location/ZLocationCell.kt [hunk 4]
```diff
@@ -33,9 +36,19 @@ class ZLocationCell : ConstraintLayout {
 
         override fun afterTextChanged(s: Editable?) {
             if (binding.textPickupLocation.hasFocus()) {
-                listener?.onLocationSearchTextChanged(s.toString(), LocationSearchFlowType.PICKUP)
+                listener?.onLocationSearchTextChanged(s.toString(), LocationSearchFlowType.PICKUP, false)
+                debounceHandler.removeCallbacksAndMessages(null)
+                debounceHandler.postDelayed({
```
Comment (rahulzoomcar) [inline zoomdls/src/main/java/com/zoomcar/uikit/location/ZLocationCell.kt:41]: We should handle debounceHandler , if the scope of using this get destroyed

File: zoomdls/src/main/java/com/zoomcar/uikit/location/ZLocationCell.kt [hunk 5]
```diff
@@ -23,6 +25,7 @@ class ZLocationCell : ConstraintLayout {
 
     private lateinit var binding: LayoutZLocationBarBinding
     private var listener: IZLocationBarListener? = null
+    private var debounceHandler = Handler(Looper.getMainLooper())
```
Comment (gideon-paul) [inline zoomdls/src/main/java/com/zoomcar/uikit/location/ZLocationCell.kt:28]: Any reason we implemented this with Handler and Looper and not coroutines?

File: zoomdls/src/main/java/com/zoomcar/uikit/location/ZLocationCell.kt [hunk 6]
```diff
@@ -33,9 +36,19 @@ class ZLocationCell : ConstraintLayout {
...
+                },
+                        1000)
             } else if (binding.textDropOffLocation.hasFocus()) {
-                listener?.onLocationSearchTextChanged(s.toString(), LocationSearchFlowType.DROP_OFF)
+                listener?.onLocationSearchTextChanged(s.toString(), LocationSearchFlowType.DROP_OFF, false)
+                debounceHandler.removeCallbacksAndMessages(null)
+                debounceHandler.postDelayed({
+                    listener?.onLocationSearchTextChanged(s.toString(), LocationSearchFlowType.DROP_OFF, true)
+                },
+                        1000)
```
Comment (gideon-paul) [inline zoomdls/src/main/java/com/zoomcar/uikit/location/ZLocationCell.kt:51]: Update with constant

**Pull request closed without merging**